File: 1458-sort-integers-by-the-number-of-1-bits/sort-integers-by-the-number-of-1-bits.cpp
class Solution {
public:
    vector<int> sortByBits(vector<int>& arr) {
        vector<pair<int,int>>pr;
        for(int i:arr)
            pr.push_back({i,__builtin_popcount(i)});
        sort(pr.begin(),pr.end(),[&](auto a,auto b)
        {
            if(a.second!=b.second)
                return a.second<b.second;
            return a.first<b.first;
        });
        vector<int>ans;
        for(auto i:pr)
            ans.push_back(i.first);
        return ans;
    }
};

File: 1458-sort-integers-by-the-number-of-1-bits/sort-integers-by-the-number-of-1-bits.java
class Pair
{
    public int first;
    public int second;

    public Pair(int first,int second)
    {
        this.first=first;
        this.second=second;
    }
}

class Solution {
    public int[] sortByBits(int[] arr) {
        Pair[] pair=new Pair[arr.length];
        for(int i=0;i<arr.length;++i)
            pair[i]=new Pair(arr[i],Integer.bitCount(arr[i]));
        Arrays.sort(pair,(a,b)->
        {
            if(a.second!=b.second)
                return a.second-b.second;
            return a.first-b.first;
        });
        int ans[]=new int[pair.length];
        for(int i=0;i<pair.length;++i)
            ans[i]=pair[i].first;
        return ans;
    }
}